### DISCUSSION TITLE
Samples will click/pop when organ is fed with short/rapid notes

### DISCUSSION THREAD

**Discussion by ray890** [GrandOrgue/grandorgue#1120] [General]
When I feed an organ manual with very short notes, mild click sounds will be generated. This happens regardless of the organ definition that I am using.  This is unrelated to the audio stream, as latency and buffer settings will not do anything to these types of pops.

I'm wondering what mechanism is at play in this process. Are samples being cut off with no fade if another instance of the same sample/note is being played? Otherwise, is there a different reason that this is happening?

If this is the situation, then it would be important to implement a (togglable?) 'failsafe' where every sample must have 4ms of fade before they are killed. This should have little impact on polyphony, and no impact on audio except for eliminating the unwanted popping.

My setup consists of each individual organ manual being allocated to each Midi Channel of my loopMIDI Port. I have found a few other situations that may cause clicking, however I suspect they click for the same reason, especially if my theory holds correct, so I will not get into them for now.

https://user-images.githubusercontent.com/11812575/166126813-b9a34ab2-b4fc-4a72-a129-baca1b044ed8.mp4

**Comment by midimusic** [comment]
Having played your on line organ (above) several times, I would not call the artefacts "clicks". They are not clicks because they do not have any significant high frequency content.

I think the notes are being stopped before they get going - what you are hearing is the beginning of the note's start transient which is not necessarily going to be particularly musical. I do not know whether the release sample gets played in these circumstances but it should not be. To find out what is really happening for very short notes you need to play just ONE very short note and examine the resulting waveform on an oscilloscope,

If this was a real organ it would probably make a similar unmusical noise when played at this speed.

csw900

**Reply by wpostma** [reply]
I do not hear clicks.  I hear very rapid cycling of samples by a human being playing more than 30 notes a second.

**Comment by willeke1234** [comment]
This might be the same issue: #306 Cutoff on very short keypress?

**Reply by ray890** [reply]
Surprisingly, it very much sounds like the same issue! I have even had the assumption that there is a bug with notes stopping during the attack if the notes are short enough

**Comment by ray890** [comment]
I had an interesting observation.  After I left GrandOrgue Demo V1 running in the background for many days, I have tested again and the clicking on ultra-short notes has disappeared, producing results in-line with what I expected, here is a recording of this. 

I don't know what exactly happened to reproduce this result.

https://user-images.githubusercontent.com/11812575/170611622-c3b9e0cf-c7cf-47ae-b951-aa8570a64b35.mp4

When I restarted the program, the clicking did end up returning. This indicates that there must be a bug in the program that is causing this.
